### PR TITLE
add test cases 

### PR DESCRIPTION
**Pull request by wzshiming**
**What this PR does / why we need it**:

Get one-letter plural name, return to empty string, resulting in generated code errors.
add some test cases with one-letter types and package names into.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

https://github.com/kubernetes/gengo/pull/117

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

File: staging/src/k8s.io/code-generator/_examples/crd/apis/test/v1/register.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	return SchemeGroupVersion.WithResource(resource).GroupResource()
+}
+
+// Adds the list of known types to api.Scheme.
+func addKnownTypes(scheme *runtime.Scheme) error {
+	scheme.AddKnownTypes(SchemeGroupVersion,
+		&T{},
+	)
+
+	scheme.AddKnownTypes(SchemeGroupVersion,
```
Comment (sttts) [inline staging/src/k8s.io/code-generator/_examples/crd/apis/test/v1/register.go]: why this? Should be part of AddToGroupVersion below.

File: staging/src/k8s.io/code-generator/_examples/crd/apis/test/v1/types.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2015 The Kubernetes Authors.
```
Comment (sttts) [inline staging/src/k8s.io/code-generator/_examples/crd/apis/test/v1/types.go]: 2018

File: staging/src/k8s.io/code-generator/_examples/crd/apis/test/v1/doc.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// +k8s:deepcopy-gen=package
+// +k8s:defaulter-gen=TypeMeta
+// +groupName=t.crd.code-generator.k8s.io
+// +groupGoName=T
+package v1
```
Comment (sttts) [inline staging/src/k8s.io/code-generator/_examples/crd/apis/test/v1/doc.go]: please use another package name than `test`, `shortnames`. `test` is reserved in Go.

File: staging/src/k8s.io/code-generator/_examples/crd/apis/test1/v1/doc.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/*
```
Comment (sttts) [inline staging/src/k8s.io/code-generator/_examples/crd/apis/test1/v1/doc.go]: use a better api group name please. shortnames, oneletter or something like that.

**Comment by wzshiming** [comment]
@sttts

**Pull request closed without merging**